roachprod: don't pollute logs with verbose ssh logs

tbg opened the pull request:
I added verbose logging in #37483 but made the poor decision to spew it
to stderr when the command exited nonzero. As a consequence, many test
failures are now obfuscated by high-verbosity ssh logs.

Change the behavior so that the error message is wrapped with a path to
the logs. We grab the contents of `~/.roachprod` in CI, so we'll be able
to access the logs in the artifacts (`roachprod_state`).

```
$ ./bin/roachprod ssh tobias-test -- echo hi
hi
$ ls ~/.roachprod/debug/
$ ./bin/roachprod ssh tobias-test -- false
Error:  ssh verbose log retained in /Users/tschottdorf/.roachprod/debug/ssh_35.231.83.24_2019-05-20T12:53:49Z: exit status 1
$ ls ~/.roachprod/debug/
ssh_35.231.83.24_2019-05-20T12:53:49Z
```

Touches #36963.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/37594)
<!-- Reviewable:end -->

nvanbenschoten commented on pkg/cmd/roachprod/hosts.go:
```diff
@@ -116,6 +116,9 @@ func loadClusters() error {
 		return err
 	}
 
+	debugDir := os.ExpandEnv(config.DefaultDebugDir)
```
nit: put this in `initHostDir` and change the name of that function.

tbg replied:
Good call, done.

nvb reviewed (approved):
Nice!

ajwerner reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> Something certainly needed to be done here, Ctrl-C on roachprod run was filling up my terminal buffer. Users may end up with a bunch of these over time but it’s probably fine. We could potentially add some logic that goes and gcs after some number of days. For another PR I suppose. 

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/37594#-:-LfKOUAV47U5nR0moxJ0:bscnu3w)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @ajwerner and @tbg)



<!-- Sent from Reviewable.io -->

tbg committed:
roachprod: don't pollute logs with verbose ssh logs

I added verbose logging in #37483 but made the poor decision to spew it
to stderr when the command exited nonzero. As a consequence, many test
failures are now obfuscated by high-verbosity ssh logs.

Change the behavior so that the error message is wrapped with a path to
the logs. We grab the contents of `~/.roachprod` in CI, so we'll be able
to access the logs in the artifacts (`roachprod_state`).

```
$ ./bin/roachprod ssh tobias-test -- echo hi
hi
$ ls ~/.roachprod/debug/
$ ./bin/roachprod ssh tobias-test -- false
Error:  ssh verbose log retained in /Users/tschottdorf/.roachprod/debug/ssh_35.231.83.24_2019-05-20T12:53:49Z: exit status 1
$ ls ~/.roachprod/debug/
ssh_35.231.83.24_2019-05-20T12:53:49Z
```

Touches #36963.

Release note: None

tbg commented:
bors r=nvanbenschoten,ajwerner
